tide/master/rest/HttpServer.cpp

```diff
@@ -47,7 +47,7 @@ using namespace zeroeq;
 std::future<http::Response> HttpServer::respondTo(http::Request& request) const
 {
     if (_blockedMethods.count(request.method) &&
-        !_isWhitelisted(request.source))
+        !_isWhitelisted(request.source.substr(0, request.source.find(":"))))
```

Comment:
the substring extraction deserves a local variable or a helper function for readability